swap_meet/vendor.py

```diff
@@ -1,2 +1,71 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        
+        return False
+    
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.remove(my_item)
+        other_vendor.add(my_item)
+
+        other_vendor.remove(their_item)
+        self.add(their_item)
+
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+        return True
+    
+    def get_by_category(self, category):
+        category_items = []
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_items.append(item)
+
+        return category_items
+    
+    def get_best_by_category(self, category):
+        best_item = None
+        condition = 0
+
+        for item in self.inventory:
+            if item.get_category() == category and item.condition > condition:
+                condition = item.condition
+                best_item = item
+        return best_item
```

Comment:
Your solution works, and it's a good solution especially since it only loops through `inventory` once! Optionally, there's a way to refactor this so it uses the `get_by_category` function and then calls `max`, if you wanted to practice that approach.